### PR TITLE
Use separate content items for licence-finders

### PR DESCRIPTION
**Pull request by binaryberry**
This publishes the start_page ("/licence-finder") to the Publishing API with an
exact route, to prepare for it to be published as a transaction by Publisher in
the future. The remaining licence-finder pages are published to the Publishing
API as content items with prefix routes.

After deploying this branch on integration, "/licence-finder/blah" returns a 404, confirming it is an exact route, and the content-id for "/licence-finder/sectors" is different from the "/licence-finder" content-id, confirming they are published as separate pieces of content.

**[DO NOT MERGE] for now as we're waiting to test this on integration tomorrow once it has fresh data.**

[Trello](https://trello.com/c/zboJadd9/409-3-republish-licence-finder-start-page-as-a-transaction-page-and-remove-hardcoded-version)

### CODE REVIEW

File: spec/notifiers/publishing_api_notifier_spec.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@
       expect(Services.publishing_api).to receive(:put_content).with('69af22e0-da49-4810-9ee4-22b4666ac627', test: :payload)
       expect(Services.publishing_api).to receive(:publish).with('69af22e0-da49-4810-9ee4-22b4666ac627', 'minor')
```
Comment (fofr) [inline spec/notifiers/publishing_api_notifier_spec.rb]: We should check here that `publishing_api` receives calls with more than just the original content id and that the payloads meet the requirements of a generic schema.

File: spec/notifiers/publishing_api_notifier_spec.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@
...
 
-      PublishingApiNotifier.publish(presenter)
+      PublishingApiNotifier.new.publish(presenter)
```
Comment (fofr) [inline spec/notifiers/publishing_api_notifier_spec.rb]: Is this changed needed?

File: app/presenters/licence_finder_form_content_item_presenter.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class LicenceFinderFormContentItemPresenter < LicenceFinderContentItemPresenter
+  attr_reader :base_path, :content_id
+
+  def initialize(base_path, content_id)
+    @base_path = base_path
+    @content_id = content_id
+  end
+
+  def update_type
```
Comment (fofr) [inline app/presenters/licence_finder_form_content_item_presenter.rb]: `update_type` can be omitted as it matches the parent class

File: app/presenters/licence_finder_content_item_presenter.rb [hunk 4]
```diff
@@ -4,13 +4,17 @@ def base_path
   end
 
   def content_id
-    metadata[:content_id]
+    "69af22e0-da49-4810-9ee4-22b4666ac627"
```
Comment (fofr) [inline app/presenters/licence_finder_content_item_presenter.rb]: Is this change necessary? The content ID should be the same for this content item so this should suffice: https://github.com/alphagov/licence-finder/blob/master/config/initializers/application_metadata.rb#L3

File: spec/notifiers/publishing_api_notifier_spec.rb (hunk 2 above)
Comment (binaryberry) [inline spec/notifiers/publishing_api_notifier_spec.rb]: That makes the spec test the `publish` instance method instead of the `self.publish` class method, which seems more relevant? The class method instantiates the presenters and calls the instance method. The latter is the one that's making the calls to the Publishing API so seems more important to test / more likely to break IMO.

File: app/presenters/licence_finder_content_item_presenter.rb (hunk 4 above)
Comment (binaryberry) [inline app/presenters/licence_finder_content_item_presenter.rb]: Good point - amended now.

File: app/presenters/licence_finder_form_content_item_presenter.rb (hunk 3 above)
Comment (binaryberry) [inline app/presenters/licence_finder_form_content_item_presenter.rb]: 👍  amended now

File: spec/notifiers/publishing_api_notifier_spec.rb (hunk 1 above)
Comment (binaryberry) [inline spec/notifiers/publishing_api_notifier_spec.rb]: 1. By _"check that publishing_api receives calls with more than just the original content id"_, do you mean also test the `self.publish` method to ensure it publishes both the start page and at least one of the other pages?

2. In regards to testing that the payloads meet the requirements of a generic schema, I feel the fact that we're already testing for that in the presenters should be enough? As long as the Presenters get instantiated the payload will meet those requirements.

Comment (binaryberry) [inline spec/notifiers/publishing_api_notifier_spec.rb]: Now amended following separate conversation and using Paul's code :-)

File: spec/notifiers/publishing_api_notifier_spec.rb (hunk 2 above)
Comment (binaryberry) [inline spec/notifiers/publishing_api_notifier_spec.rb]: Now amended following separate conversation and using Paul's code :-)
